pinot-plugins/pinot-input-format/pinot-json/src/main/java/org/apache/pinot/plugin/inputformat/json/JSONRecordExtractorUtils.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.inputformat.json;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+
+/**
+ * Helper methods for converting values from json nodes to
```

Comment:
Don't see anything related to json here. Looks like a common ExtractorUtil

Reply:
True. But right now, it is kinda specific, because it's only used by Json. only json is looking for Collection/Map/single values. Avro has its own utils which looks for Collection/Map/GenericData.Record/single value.
All others use RecordReaderUtils, which look for only Collection/single-value

Followup items:
1) Add this nested extraction support to all formats
2) In the process, unify the code in JsonRecordExtractorUtils, AvroUtils, RecordReaderUtils.

Reply:
Created the issue: https://github.com/apache/incubator-pinot/issues/5507